autotest/build_exes.py

```diff
@@ -60,131 +80,57 @@ def create_dir(pth):
-
-def build_mf6_so():
-    pm = pymake.Pymake(verbose=True)
-    pm.target = f"libmf6{soext}"
-    pm.srcdir = os.path.join("..", "srcbmi")
-    pm.srcdir2 = os.path.join("..", "src")
-    pm.appdir = os.path.join("..", "bin")
-    pm.excludefiles = [os.path.join(pm.srcdir2, "mf6.f90")]
-    pm.include_subdirs = True
-    pm.inplace = True
-    pm.makeclean = True
-
-    # reset compiler based on environmental variable, if defined
-    pm.fc = get_compiler_envvar(pm.fc)
-
-    # add strict flags if gfortran is being used
-    if pm.fc == "gfortran":
-        pm.fflags = strict_flags
-
-    # build the application
-    pm.build()
-
-    msg = f"{pm.target} does not exist."
-    assert pm.returncode == 0, msg
-
-
-def build_mf5to6():
-    # define default compilers
-    fc = "gfortran"
-    cc = None
-    fflags = None
-
-    # reset compiler based on environmental variable, if defined
-    fc = get_compiler_envvar(fc)
-
-    # determine if fortran compiler specified on the command line
+def set_compiler_environment_variable():
+    fc = None
+    # parse command line arguments
     for idx, arg in enumerate(sys.argv):
-        if arg == "-fc":
+        if arg.lower() == "-fc":
             fc = sys.argv[idx + 1]
-        elif arg in ("-ff", "--fflags"):
-            fflags = sys.argv[idx + 1]
-
-    # set source and target paths
-    srcdir = os.path.join("..", "utils", "mf5to6", "src")
-    appdir = os.path.join("..", "bin")
-    target = f"mf5to6{eext}"
-    extrafiles = os.path.join(
-        "..", "utils", "mf5to6", "pymake", "extrafiles.txt"
-    )
-
-    # build modflow 5 to 6 converter
-    pymake.main(
-        srcdir,
-        target,
-        fc=fc,
-        cc=cc,
-        fflags=fflags,
-        include_subdirs=True,
-        extrafiles=extrafiles,
-        inplace=True,
-        appdir=appdir,
-        makeclean=True,
-    )
-    target_pth = os.path.join(appdir, target)
-    msg = f"{relpath_fallback(target_pth)} does not exist."
-    assert os.path.isfile(target_pth), msg
-
-
-def build_zbud6():
-    pm = pymake.Pymake()
-    pm.target = f"zbud6{eext}"
-    pm.srcdir = os.path.join("..", "utils", "zonebudget", "src")
-    pm.appdir = os.path.join("..", "bin")
-    pm.extrafiles = os.path.join(
-        "..", "utils", "zonebudget", "pymake", "extrafiles.txt"
-    )
-    pm.inplace = True
-    pm.makeclean = True
-
-    # reset compiler based on environmental variable, if defined
-    pm.fc = get_compiler_envvar(pm.fc)
-
-    # add strict flags if gfortran is being used
-    if pm.fc == "gfortran":
-        pm.fflags = strict_flags
-
-    # build the application
-    pm.build()
-
-    msg = f"{pm.target} does not exist."
-    assert pm.returncode == 0, msg
+        elif arg.lower().startswith("-fc="):
+            fc = arg.split("=")[1]
+    # determine the FC environmental variable needs to be set
+    set_env_var = False
+    env_var = os.environ.get("FC")
+    if env_var is None:
+        set_env_var = True
+        if fc is None:
```

Comment:
Maybe you want to raise an Exception if `fc` is neither gfortran nor ifort?

Reply:
Have implemented a check. This will require future maintenance to support additional compilers (for example, `mpifort`, `mpif90`)